meson.build

```diff
@@ -75,50 +75,70 @@ configinc = include_directories('.')
 cc = meson.get_compiler('c')
 dl_dep = cc.find_library('dl', required : false)
 
-WITH_DRM = not get_option('disable_drm') and (host_machine.system() != 'windows')
-libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : (host_machine.system() != 'windows'))
-
-WITH_X11 = false
-if get_option('with_x11') != 'no'
-  x11_dep = dependency('x11', required : get_option('with_x11') == 'yes')
-  xext_dep = dependency('xext', required : get_option('with_x11') == 'yes')
-  xfixes_dep = dependency('xfixes', required : get_option('with_x11') == 'yes')
+backends = ''
+WITH_DRM = (host_machine.system() != 'windows')
+if WITH_DRM
+  backends += ' drm'
+endif
 
-  WITH_X11 = (x11_dep.found() and xext_dep.found() and xfixes_dep.found())
+# NOTE: libdrm is required by all non-win32 backends x11 via dri3, glx via x11,
+# wayland via wl_drm and drm. All of these use va_drm_util.c which need libdrm.
+libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : (host_machine.system() != 'windows'))
 
-  x11_xcb_dep = dependency('x11-xcb', required : get_option('with_x11') == 'yes')
-  xcb_dep = dependency('xcb', required : get_option('with_x11') == 'yes')
-  xcb_dri3_dep = dependency('xcb-dri3', required : get_option('with_x11') == 'yes')
-  WITH_X11 = (WITH_X11 and x11_xcb_dep.found() and xcb_dep.found() and xcb_dri3_dep.found())
+WITH_X11 = true
+if not get_option('with_x11').disabled()
```

Comment:
You have the opposite problem now, WITH_X11 is going to be true if with_x11=disabled AFAIK.

Reply:
thanks - third time's the charm :crossed_fingers: